### PR TITLE
declare support for only py>=3.5

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
it might be worth re-releasing 0.7.7 as 0.8.2 then releasing this PR as 0.9.0

**Comment by graingert** [comment]
from PEP-0592
```
One of the main scenarios where this may happen, is when dropping support for a particular version of Python. The python-requires metadata allows for dropping support for a version of Python in a way that is not disruptive to users who are still using that Python. However, a common mistake is to either omit or forget to update that bit of metadata. When that mistake has been made, a project really only has three options:

    Prevent that version from being installed through some mechanism (currently, the only mechanism is by deleting the release entirely).
    Re-release the version that worked as a higher version number, and then re-release the version that dropped support as an even higher version number with the correct metadata.
    Do nothing, and document that people using that older Python have to manually exclude that release.

With this PEP, projects can choose the first option, but with a mechanism that is less likely to break the world for people who are currently successfully using said project.
```

File: setup.py [hunk 1]
```diff
@@ -39,5 +39,6 @@
 
         ],
         ':python_version < "3.5"': 'typing >= 3.5.3',
```
Comment (mosquito) [inline setup.py]: it's redundant probably.

Comment (graingert) [inline setup.py]:
```suggestion
```

File: setup.py [hunk 2]
```diff
@@ -38,6 +38,6 @@
             'xmltodict',
 
         ],
-        ':python_version < "3.5"': 'typing >= 3.5.3',
-    }
+    },
+    python_requires='>=3.5',
```
Comment (graingert) [inline setup.py]:
```suggestion
    python_requires=">='3.5'",
```

Comment (graingert) [reply setup.py]: no the original was correct https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

**Pull request merged**